plugins/content_distribution/providers/tvinci/lib/api/KalturaTvinciDistributionJobProviderData.php

```diff
@@ -95,24 +95,50 @@ public function __construct(KalturaDistributionJobData $distributionJobData = nu
 
 	}
 
-	private function initPlayManifestUrls($entry, $feedHelper)
+	private static function getVideoAssetDataMap()
 	{
-		$videoAssetDataMap = array(
-			array( 'Main',						PlaybackProtocol::AKAMAI_HDS,	'mbr',		'a4m' ),
-			array( 'Tablet Main',				PlaybackProtocol::APPLE_HTTP,	'ipad',		'm3u8' ),
-			array( 'Smartphone Main',			PlaybackProtocol::APPLE_HTTP,	'iphone',	'm3u8' ),
+		return array(
+			array( 'Main',						PlaybackProtocol::AKAMAI_HDS,	array('mbr'),		    		'a4m' ),
+			array( 'Tablet Main',				PlaybackProtocol::APPLE_HTTP,	array('ipadnew','ipad'),		'm3u8' ),
+			array( 'Smartphone Main',			PlaybackProtocol::APPLE_HTTP,	array('iphonenew','iphone'),	'm3u8' ),
 		);
+	}
 
-		// Loop and build the file nodes
+	private function createPlayManifestURLs(KalturaEntryDistribution $entryDistribution, entry $entry, TvinciDistributionFeedHelper $feedHelper)
+	{
+		$distributionFlavorAssets  = assetPeer::retrieveByIds(explode(',', $entryDistribution->flavorAssetIds));
+		$videoAssetDataMap = $this->getVideoAssetDataMap();
 		foreach ( $videoAssetDataMap as $videoAssetData )
 		{
 			$tvinciAssetName = $videoAssetData[0];
 			$playbackProtocol = $videoAssetData[1];
-			$tag = $videoAssetData[2];
+			$tags = $videoAssetData[2];
 			$fileExt = $videoAssetData[3];
-			$fileCoGuid = "{$entry->getIntId()}{$entry->getFlavorParamsIds}";
-			$url = $this->getPlayManifestUrl($entry, $playbackProtocol, $tag, $fileExt);
-			$feedHelper->setVideoAssetData( $tvinciAssetName, $url,$fileCoGuid );
+			$keys = array();
+			$relevantTags = array();
+			foreach ( $distributionFlavorAssets as $distributionFlavorAsset )
+			{
+				foreach ($tags as $tag)
+				{
+					if ($distributionFlavorAsset->isLocalReadyStatus() &&
+						$distributionFlavorAsset->hasTag($tag) )
+					{
+						$keys[] = "{$entry->getEntryId()}_{$distributionFlavorAsset->getFlavorParamsId()}";
```

Comment:
Please refactor -
Create static func that gets entryid,flavour param and returns the correct string.